Task-WebRTC.md

```diff
@@ -0,0 +1,370 @@
+
+## Message States (Beyond 'auth')
+
+```
+           +----------+
+           |          |
+    +------+----------v------+    +-------+
+    | offer/answer/candidate +--->+ close |
+    +------------------------+    +-------+
+```
+
+## Message Flow Example (Beyond 'auth')
+
+```
+    Initiator                 Responder
+     |                               |
+     |             offer             |
+     |------------------------------>|
+     |             answer            |
+     |<------------------------------|
+     |      candidate (n times)      |
+     |<----------------------------->|
+     |             close             |
+     |<----------------------------->|
+     |                               |
+```
+
+## 'offer' Message
+
+At any time, the initiator MAY send an 'offer' message to the 
+responder.
+
+The initiator MUST set the *offer* field to the dictionary/object of 
+its WebRTC `RTCPeerConnection`'s local description it has generated 
+with calling `createOffer` on the `RTCPeerConnection` instance. The 
+*offer* field SHALL be a dictionary/object and MUST contain:
+
+* The *type* field containing a valid `RTCSdpType` in string 
+  representation.
+* The *sdp* field containing a blob of SDP data in string 
+  representation. If the *type* field is `rollback`, the field MAY be 
+  omitted.
+
+The responder SHALL validate that the *offer* field is a dictionary/
+object containing the above mentioned fields and value types. It SHALL 
+continue by setting the value of that field as the WebRTC 
+`RTCPeerConnection`'s remote description and creating an answer.
+
+The message SHALL be NaCl public-key encrypted by the client's 
+session key pair and the other client's session key pair.
+
+## 'answer' Message
+
+Once the responder has set the remote description on its WebRTC 
+`RTCPeerConnection` instance and generated an answer by calling 
+`createAnswer` on the instance, it SHALL send an 'answer' message. The 
+*answer* field SHALL be a dictionary/object and MUST contain:
+
+* The *type* field containing a valid `RTCSdpType` in string 
+  representation.
+* The *sdp* field containing a blob of SDP data in string 
+  representation. If the *type* field is `rollback`, the field MAY be 
+  omitted.
+  
+The initiator SHALL validate that the *answer* field is a dictionary/
+object containing the above mentioned fields and value types. It SHALL 
+continue by setting the value of that field as the WebRTC 
+`RTCPeerConnection`'s remote description.
+
+The message SHALL be NaCl public-key encrypted by the client's 
+session key pair and the other client's session key pair.
+
+## 'candidate' Message
+
+Both clients MAY send ICE candidates at any time to each other. 
+Clients SHOULD bundle available candidates.
+
+A client who sends an ICE candidate SHALL set the *candidate* field to 
+a list/an array of dictionaries/objects where each dictionary/object 
+SHALL contain the following fields:
```

Comment:
The example that has been added in the meantime contains a map as value for `candidate`, not an array.